apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -40,97 +41,95 @@ export class FileService {
-      email: answers.email as string,
-      address: nationalRegistryData.address?.streetAddress as string,
-      postalCode: nationalRegistryData.address?.postalCode as string,
-      city: nationalRegistryData.address?.city as string,
+  async uploadSignedFile(
+    application: Application,
+    documentToken: string,
+    type: PdfTypes,
+  ) {
+    const bucket = this.config.presignBucket
+    if (!bucket) {
+      throw new Error(' Bucket not configured.')
     }
 
-    const expiry = answers.expiry as string
-
-    return { parentA, parentB, childrenAppliedFor, expiry }
+    await this.signingService
+      .getSignedDocument(DokobitFileName[type], documentToken)
+      .then((file) => {
+        const s3FileName = `${BucketTypePrefix[type]}/${application.id}.pdf`
+        uploadFile(Buffer.from(file, 'binary'), bucket, s3FileName)
+      })
   }
 
-  private async createResidenceChangePdf(
-    parentA: ParentResidenceChange,
-    parentB: ParentResidenceChange,
-    childrenAppliedFor: Array<PersonResidenceChange>,
-    expiry: string,
-    applicationId: string,
-  ): Promise<string> {
-    const pdfBuffer = await generateResidenceChangePdf(
-      childrenAppliedFor,
-      parentA,
-      parentB,
-      expiry,
-    )
-
-    const fileName = `children-residence-change/${parentA.ssn}/${applicationId}.pdf`
-    const bucket = this.config.presignBucket || ''
+  async requestFileSignature(
+    application: Application,
+    type: PdfTypes,
+  ): Promise<SigningServiceResponse> {
+    const answers = application.answers as FormValue
+    const externalData = application.externalData as FormValue
 
-    return await this.getPresignedUrl(pdfBuffer, bucket, fileName)
+    switch (type) {
+      case PdfTypes.CHILDREN_RESIDENCE_CHANGE: {
+        const { phoneNumber, name } = applicantData(answers, externalData)
+        return await this.handleChildrenResidenceChangeSignature(
+          type,
+          application.id,
+          name,
+          phoneNumber,
+        )
+      }
+    }
   }
 
-  private async getPresignedUrl(
-    buffer: Buffer,
-    bucket: string,
-    fileName: string,
-  ): Promise<string> {
-    const uploadParams = {
-      Bucket: bucket,
-      Key: fileName,
-      ContentEncoding: 'base64',
-      ContentDisposition: 'inline',
-      ContentType: 'application/pdf',
-      Body: buffer,
+  private async handleChildrenResidenceChangeSignature(
+    type: PdfTypes,
+    applicationId: string,
+    applicantName: string,
+    phoneNumber?: string,
+  ): Promise<SigningServiceResponse> {
+    const bucket = this.config.presignBucket
+    if (!bucket) {
+      throw new Error(' Bucket not configured.')
     }
 
-    await this.s3
-      .upload(uploadParams)
-      .promise()
-      .catch(() => {
-        return null
-      })
+    const s3FileName = `${BucketTypePrefix[type]}/${applicationId}.pdf`
+    const s3File = await getFile(bucket, s3FileName)
+    const fileContent = s3File.Body?.toString('binary')
```

Comment:
We should use buffers instead of binary strings whenever possible.

Reply:
Yes agree. 
When implementing this I tried that approach and when you rightfully poked at this then I tried at again but no luck 😭 
The problem I think is the way dokobit works. If I send anything else then a binary string to them then I get an error from their side or and empty document. I talked to @gudjong  about this since he implemented the signing service and he had the same experience. If you have any suggestions to try out them I am all ears. 🙏

Reply:
I'm just talking about it on our end. 

The dokubit nest service is doing `Base64.btoa(string)` before sending the document to their API. So instead of us converting buffers to binary strings to base64, we can just pass buffers around and then do `buffer.toString('base64')` which should at least save one copy of the file in memory, and the base64 encoding probably runs in native code.

Reply:
Yes totally agree on paper. Tried this out at first and then did it again now to make sure I wasn't doing something silly. This is the response I get from Dokobit when I try that: 
`Uploaded document is not compliant with PDF standard` 🤷

Reply:
Whaaaaaat??? 🤯